WIP: 2.0

connor-baer opened the pull request:
# DONT MERGE THIS

This branch is a **work in progress** of the new brand design that will be used starting with the new [component library](https://github.com/mlent/component-library). 
I've also updated some dependencies, most notably Bourbon to 5.0 and Neat to 2.0. Have a look at the [Bourbon 5.0](http://bourbon.netlify.com/docs/latest/) and [Neat 2.0](http://neat.bourbon.io/docs/2.0.0/) docs.

### TODO

- [x] Update to Bourbon 5.0 and Neat 2.0 and fix deprecation warnings
- [x] Update to new design based on the [fabulous work](https://share.abstractapp.com/64ada0c7-9069-4f7a-b632-1204dbc8e739) of the design team
- [x] Lint using [stylelint](https://stylelint.io)
- [x] Update README
- [ ] Include updates from `master`

connor-baer committed:
Transfer changes from component branch

connor-baer committed:
Upgrade to Neat 2.0

Neat 2.0 is a complete rewrite. All old mixins were removed and replaced
by 7 new ones. I wrote a custom breakpoints mixin and removed all
deprecated mixins. Vendor-prefixing is now handled by autoprefixer.
See the documentation for more: http://neat.bourbon.io/docs/2.0.0/

connor-baer committed:
Improve breakpoint mixin

mollidor commented on package.json:
```diff
@@ -9,16 +9,16 @@
   "repository": {
     "type": "git",
-    "url": "git+https://github.com/felixjung/brand-styles.git"
+    "url": "git+https://github.com/sumup/brand-styles.git"
```
😄

felixjung replied:
😆

connor-baer committed:
Add neat to dependencies

connor-baer committed:
Clean up and restructuring

I removed unused and superflous code and restructured the files.

connor-baer committed:
Further cleanup

connor-baer committed:
Remove vendor-prefixing mixins

Bourbon will remove all vendor-prefixing mixins in version 5. Besides,
autoprefixer should (and is) used to add vendor prefixes.

connor-baer committed:
Restructure code, new design

Extract mixins into a new folder since the `_mixins.scss` file got too
big.
Update colors, buttons and text styles to new design.
Remove disabled button hover styles, since `pointer-events: none` takes
care of that.
Remove some superflous vendor prefixes.

connor-baer committed:
Add alignment classes

connor-baer committed:
Update and improve mixins

connor-baer committed:
Update drop-shadow mixin

connor-baer committed:
Add box-sizing reset

connor-baer committed:
Update line heights

connor-baer committed:
One grid to rule them all

connor-baer committed:
Update shadows

connor-baer committed:
New brand colors

connor-baer committed:
Add $spacing-xxxl: 60px

Sections in the new component library are spaced above and below
 (120px from each other).

connor-baer committed:
Reduce button padding

connor-baer committed:
Change button size again

connor-baer committed:
Add background gradients

connor-baer committed:
Add gutter variable

connor-baer committed:
Add MuseoSans 900 font

connor-baer committed:
Switch to relative media-queries

I've updated the media queries and `media` mixin to use `em`s as unit.
Read why: http://bradfrost.com/blog/post/7-habits-of-highly-effective-media-queries/#relative
TL;DR: It optimizes the presentation of content when the user has set
a different base font size or zooms the page.

connor-baer committed:
Add theming to hamburger

connor-baer committed:
Add btn--outline

connor-baer committed:
Increase font weights

connor-baer committed:
Tweak breakpoints

connor-baer committed:
Final breakpoints

felixjung commented on _variables.scss:
```diff
@@ -89,6 +89,15 @@ $breakpoints: (
 );
 
+// TODO:👇 Deprecated (in use on dashboard)
+$until-medium-screens: new-breakpoint(max-width 450px 4) !default;
```
Yo, FYI, dashboard has it's own bad breakpoints 😁 These are only used in static-marketing.

connor-baer replied:
Thanks!

connor-baer committed:
Animate links on hover

JulianeAlbuquerque committed:
Add new gradient vars to badge

JulianeAlbuquerque committed:
Merge branch '2.0' of https://github.com/sumup/brand-styles into 2.0

JulianeAlbuquerque committed:
Rename variables to pricingBadge

connor-baer committed:
Improve buttons

- Active state for buttons
- Improve outlined button
- Minimum width for buttons: 120px;

connor-baer committed:
Remove font-weight: 700

connor-baer committed:
Darker base color

connor-baer committed:
Update colors

connor-baer committed:
Revert blue gradient

connor-baer committed:
Prevent horizontal scrolling on body

connor-baer committed:
Add spacing between adjacent buttons

connor-baer committed:
Update secondary button color

connor-baer committed:
Reduce height of outlined button

connor-baer committed:
Configure stylelint for linting

Replaces scss-lint

connor-baer committed:
Fix most style-lint warnings

connor-baer committed:
Update .btns to new design

connor-baer committed:
Fix spacing on grouped buttons

mlent committed:
Style form inputs to new design.

connor-baer committed:
Update text button styles

Commit:
Break long words on mobile

connor-baer committed:
[MW-689]Update grid spacing.

connor-baer committed:
Remove background on hover from text buttons.

Commit:
[MW-755] Remove hyphens from BR mobile sites

mlent committed:
[MW-724] Style other form inputs (textarea) for POS forms.

connor-baer committed:
[MW-707] Reduce large button radius.

connor-baer committed:
[MW-707] Tweak button hover and active states.

connor-baer committed:
MW-812 Update Bulgarian font to Roboto.

connor-baer committed:
[MW-812] Update medium Roboto font weight.

connor-baer committed:
Allow grid config overrides.

mlent committed:
[MW-744] Add outline style for inputs.

mlent committed:
[MW-744] Fix a typography inconsistency.

connor-baer committed:
[MW-848] Remove default styling from buttons.

connor-baer committed:
[MW-919] Add Chilean brand color.

connor-baer committed:
fixup! [MW-919] Add Chilean brand color.

connor-baer committed:
Update button drop shadows to fit CL brand color

connor-baer commented:
Closing, since we've moved to [⚡Circuit UI](https://github.com/sumup/circuit-ui) since.
